skyportal/tests/api/test_annotations.py

```diff
@@ -0,0 +1,296 @@
+import numpy.testing as npt
+from skyportal.tests import api
+
+
+def test_post_without_origin_fails(annotation_token, public_source, public_group):
+
+    # this should not work, since no "origin" is given
+    status, data = api(
+        'POST',
+        'annotation',
+        data={
+            'obj_id': public_source.id,
+            'data': {'redshift': 0.15},
+            'group_ids': [public_group.id],
+        },
+        token=annotation_token,
+    )
+
+    assert status == 400
+    assert data["message"] == "Missing required field `origin`"
+
+    # this should not work, since "origin" is empty
+    status, data = api(
+        'POST',
+        'annotation',
+        data={
+            'obj_id': public_source.id,
+            'origin': '',
+            'data': {'redshift': 0.15},
+            'group_ids': [public_group.id],
+        },
+        token=annotation_token,
+    )
+
+    assert status == 400
+    assert data["message"] == "Missing required field `origin`"
+
+
+def test_post_same_origin_fails(annotation_token, public_source, public_group):
+
+    # first time adding an annotation to this object from Kowalski
+    status, data = api(
+        'POST',
+        'annotation',
+        data={
+            'obj_id': public_source.id,
+            'origin': 'kowalski',
+            'data': {'redshift': 0.15},
+            'group_ids': [public_group.id],
+        },
+        token=annotation_token,
+    )
+
+    assert status == 200
+
+    # this should not work, since "origin" Kowalski was already posted to this object
+    # instead, try updating the existing annotation if you have new information!
+    status, data = api(
+        'POST',
+        'annotation',
+        data={
+            'obj_id': public_source.id,
+            'origin': 'kowalski',
+            'data': {'redshift': 0.15},
+            'group_ids': [public_group.id],
+        },
+        token=annotation_token,
+    )
+
+    assert status == 400
+    assert 'duplicate key value violates unique constraint' in data["message"]
```

Comment:
I would refrain from checking the messages as this may increase the brittleness of the test suite but I defer to @stefanv  on this if he thinks otherwise